src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1155,14 +1156,26 @@ protected override void HandleContinueRequestAsync(IRequestResponder<ContinueArg
             }
         }
 
-        protected override void HandleStepInRequestAsync(IRequestResponder<StepInArguments> responder)
+        protected override void HandleReverseContinueRequestAsync(IRequestResponder<ReverseContinueArguments> responder)
         {
-            StepInResponse response = new StepInResponse();
+            try
+            {
+                // FIXME: there is no IDebugProgram2.ReverseContinue
```

Comment:
We have [`IDebugReversibleEngineProgram160.SetExecuteDirection`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.debugger.interop.idebugreversibleengineprogram160?view=visualstudiosdk-2019)

For every ReverseRequest we set `SetExecuteDirection` to `backward` and for every regular continue/step request we set it to `foward`.

We will also need to add a reference to `Microsoft.VisualStudio.Debugger.Interop.16.0`

Reply:
But how does it interact with `STEPKIND.STEP_BACKWARDS`?

Reply:
When `STEPKIND.STEP_BACKWARDS` was created, it did not take into the idea of  `step in`, `step out`, and `step over`. We created the IDebugReversibleEngine and Session to address this issue. 

I would remove `STEPKIND.STEP_BACKWARDS`, since even in the Visual Studio Debugger, we dismiss that enum.